### PR TITLE
ref(autofix): Agent refactor attempt 2

### PR DESCRIPTION
**Pull request by jennmueng**
Tested thoroughly via local e2e getsentry setup

1. Root cause step
2. Fix step
3. Fix step after waiting 10 minutes
4. Rethinking
5. Interjections
6. Multiple interjections
7. Spamming rethinks and interjections
8. fix feedback
9. Unit test generation
10. code diff editing
11. codecov pr unit test gen
12. pr creation #1587 

Rethink after 10 minutes to be fixed in a separate pr: https://github.com/orgs/getsentry/projects/171/views/7?pane=issue&itemId=90308927

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,3 +140,20 @@ accidentally include any sensitive personal files in your source tree before usi
...
 allowing you to track when the server has loaded that release version.
+
+
+## Running Tests
+
+You can run all tests with `make test`.
+
+Make sure you have the test database running when running individual tests, do that via `docker compose up -d test-db`.
+
+To run a single test, use `pytest tests/path/to/test.py::test_name`.
```
Comment (roaga) [inline README.md:151]: do you not need to run `make shell` first anymore?

File: src/seer/automation/autofix/components/insight_sharing/component.py [hunk 2]
```diff
@@ -5,24 +5,29 @@
...
         latest_thought: str,
         past_insights: list[str],
         step_type: str | None = None,
     ):
+        no_insight_instruction = (
+            "If there is no clear conclusion that adds a ton of new insight and value, return <NO_INSIGHT/>. If it is similar to a previous conclusion, return <NO_INSIGHT/>."
+            if past_insights
+            else ""
+        )
+
```
Comment (roaga) [inline src/seer/automation/autofix/components/insight_sharing/component.py:30]: are these prompts up to date with `main`?

**Review by roaga — APPROVED** [review]
It looks good, but I'm scared

File: README.md (hunk 1 above)
Comment (jennmueng) [reply README.md:151]: you're right, i missed a step

**Pull request merged**